h/views/oidc.py

```diff
@@ -121,7 +135,7 @@ def __init__(self, request: Request) -> None:
         self._orcid_client = request.find_service(ORCIDClientService)
         self._user_service = request.find_service(name="user")
 
-    @view_config(is_authenticated=True)
+    @view_config()
```

Comment:
The `redirect()` view now needs to be callable when not logged in.

TODO: Need to handle the case where `action="connect"` but the user is not logged in. It should show an error page.

Reply:
What is the state of these TODOs? If not done, are they tracked somewhere?

Reply:
They are tracked here! I forgot about them but I'll fix them before merging this

Reply:
Done: https://github.com/hypothesis/h/pull/9707/commits/8a48da4f0f93ab97b4e95b53e4ad3dbb998057b2